app/src/main/java/ch/deletescape/lawnchair/weather/owm/OWMWeatherDownload.java

```diff
@@ -0,0 +1,100 @@
+package ch.deletescape.lawnchair.weather.owm;
+
+import android.os.AsyncTask;
+import android.support.annotation.Nullable;
+import android.util.Log;
+
+import org.json.JSONObject;
+
+import java.io.ByteArrayOutputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+import ch.deletescape.lawnchair.weather.WeatherUnits;
+
+public class OWMWeatherDownload {
+
+    private String TAG = "OWMWeatherDownload";
+
+    private String apiKey;
+    private String location;
+    private WeatherUnits weatherUnits;
+    private OWMWeatherCallback weatherCallback;
+
+    public OWMWeatherDownload setAPIKey(String apiKey) {
+        this.apiKey = apiKey;
+        return this;
+    }
+
+    public OWMWeatherDownload setLocation(String location) {
+        this.location = location.replaceAll(" ", "");
+        return this;
+    }
+
+    public OWMWeatherDownload setWeatherUnits(WeatherUnits weatherUnits) {
+        this.weatherUnits = weatherUnits;
+        return this;
+    }
+
+    public OWMWeatherDownload download(OWMWeatherCallback weatherCallback) {
+        this.weatherCallback = weatherCallback;
+        new WeatherDownload().execute();
+        return this;
+    }
+
+    private class WeatherDownload extends AsyncTask<Void, Void, Void> {
+        JSONObject json = null;
+        @Override
+        protected Void doInBackground(Void... voids) {
+            json = download();
+            return null;
+        }
+        @Override
+        protected void onPostExecute(Void aVoid) {
+            super.onPostExecute(aVoid);
+            if (json != null)
+                new OWMWeatherData(json, weatherCallback, weatherUnits);
+            else
+                weatherCallback.onFailed();
+        }
+    }
+
+    @Nullable
+    private JSONObject download() {
+
+        String mURL;
+
+        if (weatherUnits == WeatherUnits.IMPERIAL)
+            mURL = "http://api.openweathermap.org/data/2.5/weather?q=" + location + "&units=imperial&APPID="
+                    + apiKey;
+        else
+            mURL = "http://api.openweathermap.org/data/2.5/weather?q=" + location + "&units=metric&APPID="
+                    + apiKey;
```

Comment:
if (weatherUnits == WeatherUnits.IMPERIAL)
             mURL = "http://api.openweathermap.org/data/2.5/weather?q=" + location + "&units=imperial&APPID="
                   + apiKey;
         else
            mURL = "http://api.openweathermap.org/data/2.5/weather?q=" + location + "&units=metric&APPID="
                    + apiKey;

i think better it will be look like 
   mURL = "http://api.openweathermap.org/data/2.5/weather?q=" + location + "&units="+weatherUnits +"&APPID="
                   + apiKey;